packages/js/product-editor/src/hooks/use-clear-selected-block-on-blur/use-clear-selected-block-on-blur.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * External dependencies
+ */
+import { useDispatch } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+// This is a workaround to hide the toolbar when the block is blurred.
+// This is a temporary solution until using Gutenberg 18 with the
+// fix from https://github.com/WordPress/gutenberg/pull/59800
+export const useClearSelectedBlockOnBlur = () => {
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	// @ts-ignore No types for this exist yet.
+	const { clearSelectedBlock } = useDispatch( blockEditorStore );
+
+	function handleBlur( event: {
+		relatedTarget: ( EventTarget & Element ) | null;
+	} ) {
+		const isToolbar = event.relatedTarget?.closest(
```

Comment:
I'm slightly concerned about not checking the related target property. Although it expects to be defined, would check whether it's defined prevent unexpected errors?

```suggestion
		const isToolbar = event?.relatedTarget?.closest(
```